# DSCI_100 Individual Project Planning


GitHub link: https://github.com/Alvin-Fong/DSCI100-Project.git

**Importing Data**

In [10]:
#importing library
library(tidyverse)

#reading the dataframe from URL
url_players = "https://drive.google.com/uc?rxport-download&id=10gyqmLlK-jPguz2n4gdqru3SiHnwC33C"
players <- read_csv(url_players)
head(players, n=5)

url_sessions = "https://drive.google.com/uc?rxport-download&id=1SiXmFsWrbG81jK3z3feJCmMYCDmr-f1y"
sessions <- read_csv(url_sessions)
head(sessions, n=5)


#data descriptions 
distinct(players, experience)
distinct(players, subscribe)
distinct(players, gender)

players |> summarize(max(played_hours),min(played_hours), max(Age, na.rm = TRUE), min(Age, na.rm = TRUE))
sessions |> summarize(max(original_start_time),min(original_start_time), max(original_end_time, na.rm = TRUE), 
                      min(original_end_time, na.rm = TRUE))

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21

Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


hashedEmail                                                     
1 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
  start_time       end_time         original_start_time original_end_time
1 30/06/2024 18:12 30/06/2024 18:24 1.71977e+12         1.71977e+12      
2 17/06/2024 23:33 17/06/2024 23:46 1.71867e+12         1.71867e+12      
3 25/07/2024 17:34 25/07/2024 17:57 1.72193e+12         1.72193e+12      
4 25/07/2024 03:22 25/07/2024 03:58 1.72188e+12         1.72188e+12      
5 25/05/2024 16:01 25/05/2024 16:12 1.71665e+12         1.71665e+12

experience
1 Pro       
2 Veteran   
3 Amateur   
4 Regular   
5 Beginner

subscribe
1  TRUE    
2 FALSE

gender           
1 Male             
2 Female           
3 Non-binary       
4 Prefer not to say
5 Agender          
6 Two-Spirited     
7 Other

max(played_hours) min(played_hours) max(Age, na.rm = TRUE)
1 223.1             0                 58                    
  min(Age, na.rm = TRUE)
1 9

max(original_start_time) min(original_start_time)
1 1.72733e+12              1.7124e+12              
  max(original_end_time, na.rm = TRUE) min(original_end_time, na.rm = TRUE)
1 1.72734e+12                          1.7124e+12

### 1) Data Description

The **`players`** features **196** observations with **7** variables. 
    
Variables:
+ **`experience`**: character variable, five categories for the experience level of the player: 'Pro', 'Veteran', 'Amateur', 'Regular', 'Beginner'.
+ **`subscribe`**: logical variable, two categories for if the player is subscribed or not, 'True', 'False'.
+ **`hashedEmail`**: character variable, records the hashed email of the player
+ **`played_hours`**: double variable, records the hours logged by the player, hours from 0 to 223.1.
+ **`name`**: character variable, records the name of the players
+ **`gender`**: character variable, seven categories of the gender of the player, 'Male', 'Female', 'Non-binary', 'Prefer not to say', 'Agender', 'Two-Spirited', 'Other'
+ **`Age`**: double variable, records the age of the player, ages from 9 to 58.


The **`sessions`** features **1535** observations with **5** variables. 

Variables:
+ **`hashedEmail`**: character variable, records the hashed email of the player
+ **`start_time`**: character variable, records the start date and time of the player
+ **`end_time`**: character variable, records the end date and time of the player
+ **`original_start_time`**: double variable, records the original start time of the player, from 1.71e+12 to 1.73e+12.
+ **`original_end_time`**: double variable, records the original end time of the player, from 1.71e+12 to 1.73e+12.

#### Potential Issues:
+ missing ages
+ measurement error(eg. age/hours inaccuracies)
+ time away from keyboard/idle time(eg. program running the background)

### 2) Question
Can we use **age** and **played hours** to determine the **experience** of the player?

Using the **`players`** dataset, I will use the `age` and `played hours` variables as predictors to predict the experience of the player. I will use a scatterplot to determine if there is a relationship between these variables and if there is, to futher explore the strength of this relationship. Due to the fact that I am predicting a label, I will use knn classification as the predicitive method. To wrangle the data for knn classification, I would need to first convert the experience variable from a chracter variable to a factor variable.

### 3) Exploratory Data Analysis and Visualization
#### Players:
**`Players`** does not require wrangling as the data is already tidy. This is because each observation is a row, and each column corresponds to a variable. Each cell also features only one value.

#### Sessions:
**`Sessions`** does require wrangling as the data is not tidy; however, the dataset is not required to answer the question, therefore, wrangling is unnecessary. 

In [47]:
players |> summarize(across(c(played_hours, Age),~mean(.x,na.rm = TRUE)))

played_hours Age     
1 5.845918     21.13918

|**played_hours**|**Age**|
|--- | --- |
| 5.85| 21.14|
